### PR TITLE
Browser nav bar state update

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-921

## Description
- Updated the Navbar open/close sate to be species independent
- Updated Browser.test.tsx file to use react testing library

## Deployment URL
http://navigation-panel-state.review.ensembl.org

## Views affected
Browser -> NavBar

### Other effects
- [ ] I have checked any requirements for Google Analytics
- [x] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

### CODE REVIEW

File: src/ensembl/src/content/app/browser/Browser.test.tsx [hunk 1]
```diff
@@ -70,43 +76,45 @@ describe('<Browser />', () => {
...
-      const wrapper = mountBrowserComponent();
+      let { container } = mountBrowserComponent();
 
-      expect(wrapper.find(BrowserImage)).toHaveLength(0);
-      expect(wrapper.find(TrackPanel)).toHaveLength(0);
+      expect(container.querySelectorAll('.browserImage')).toHaveLength(0);
+      expect(container.querySelectorAll('.trackPanel')).toHaveLength(0);
 
-      wrapper.setProps({
+      container = mountBrowserComponent({
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/Browser.test.tsx:112]: The direct equivalent to enzyme's `setProps` is the `rerender` method on the object returned from the `render` function. Don't know if it's better any better here than redefining the container; but just thought I'd mention this possibility.

File: src/ensembl/src/content/app/browser/Browser.tsx [hunk 2]
```diff
@@ -169,7 +169,7 @@ export const ExampleObjectLinks = (props: BrowserProps) => {
   });
 
   return (
-    <div>
+    <div data-test-id="example links">
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/Browser.tsx:172]: You aren't using it anywhere, are you?

File: src/ensembl/src/content/app/browser/browser-image/BrowserImage.tsx [hunk 3]
```diff
@@ -68,7 +78,7 @@ export type BrowserImageProps = {
 };
 
 type BpaneOutPayload = {
-  bumper?: BrowserNavStates;
+  bumper?: boolean[];
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browser-image/BrowserImage.tsx:81]: This type has now become too unspecific — we are losing both the information that there are six of these booleans here, and what these booleans represent.

A labelled tuple type would be a better fit:

```
[
  top: boolean,
  right: boolean,
  bottom: boolean,
  left: boolean,
  zoomOut: boolean,
  zoomIn: boolean
]
```

it's up to you whether you want to define this type here inline (ok if we are not reusing it) or define it as a separate type (might help if we are reusing it, e.g. when we are generating a payload to send back to genome browser)

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx [hunk 4]
```diff
@@ -15,49 +15,64 @@
...
+        browserNavIconStates={browserNavIconStates as BrowserNavIconStates}
         isDisabled={true}
       />
     );
     expect(container.querySelector('.overlay')).toBeTruthy();
   });
 
   it('disables buttons if corresponding actions are not possible', () => {
-    // browserNavStates are an array of booleans that indicate whether the button
+    // browserNavIconStates are an` array of booleans that indicate whether the button
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx:62]: extra tick after `an`?

File: src/ensembl/src/content/app/browser/browserActions.ts [hunk 5]
```diff
@@ -224,18 +224,27 @@ export const restoreBrowserTrackStates: ActionCreator<ThunkAction<
...
 export const closeBrowserNav = createAction(
-  'browser/close-browser-navigation'
-)();
+  'browser/close-browser-navigation',
+  (activeGenomeId: string) => {
+    return {
+      activeGenomeId
+    };
+  }
+)<{ activeGenomeId: string }>();
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browserActions.ts:247]: what's the purpose of this change?

File: src/ensembl/src/content/app/browser/browserState.ts [hunk 6]
```diff
@@ -23,15 +23,28 @@ const activeEnsObjectIds = browserStorageService.getActiveEnsObjectIds();
 const trackStates = browserStorageService.getTrackStates();
 const chrLocations = browserStorageService.getChrLocation();
 
+export enum BrowserNavIcons {
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browserState.ts:26]: If it's an enum, it's describing a single thing at a time; therefore it should be singular rather than plural. Icon, not Icons. Maybe even Action rather than Icon (icon is too much of a concrete UI image thing).

File: src/ensembl/src/content/app/browser/Browser.test.tsx (hunk 1 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/Browser.test.tsx]: Updated it to use `rerender`

File: src/ensembl/src/content/app/browser/Browser.tsx (hunk 2 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/Browser.tsx]: Yes, that's right. Removed it.

File: src/ensembl/src/content/app/browser/browser-image/BrowserImage.tsx (hunk 3 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browser-image/BrowserImage.tsx]: I have defined it as a separate type for now 👍

File: src/ensembl/src/content/app/browser/browserState.ts (hunk 6 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browserState.ts]: That's done

File: src/ensembl/src/content/app/browser/browserActions.ts (hunk 5 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browserActions.ts:242]: I don't that there is any good purpose behind this change. I have reverted it.

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx (hunk 4 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx]: Remove it 👍

File: src/ensembl/src/content/app/browser/Browser.test.tsx [hunk 7]
```diff
@@ -116,23 +127,28 @@ describe('<Browser />', () => {
...
-          browserNavOpened: true
+          browserNavOpenState: true
         });
-        expect(wrapper.find(BrowserNavBar).length).toBe(1);
 
-        wrapper.setProps({ isDrawerOpened: true });
-        expect(wrapper.find(BrowserNavBar).length).toBe(0);
+        expect(container.querySelectorAll('.browserNavBar')).toHaveLength(1);
+
+        container = mountBrowserComponent({ isDrawerOpened: true }).container;
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/Browser.test.tsx:149]: Will this test also benefit from rerender?

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx [hunk 8]
```diff
@@ -15,49 +15,64 @@
...
+        browserNavIconStates={browserNavIconStates as BrowserNavIconStates}
         isDisabled={true}
       />
     );
     expect(container.querySelector('.overlay')).toBeTruthy();
   });
 
   it('disables buttons if corresponding actions are not possible', () => {
-    // browserNavStates are an array of booleans that indicate whether the button
+    // browserNavIconStates are a set of booleans that indicate whether the button
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx:62]: `BrowserNavIconStates` isn't a set really, is it? It's a hash map. But after your changes I suppose these two lines of comment can go away; it's now much clearer from the types what `browserNavIconStates` are

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx [hunk 9]
```diff
@@ -15,49 +15,64 @@
...
     // has already caused maximum corresponding effect, and will have no further effect if pressed
-    const { container } = render(
+    render(
       <BrowserNavBarControls
-        browserNavStates={browserNavStates}
+        browserNavIconStates={browserNavIconStates as BrowserNavIconStates}
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx:66]: Instead of writing `as BrowserNavIconStates` here, we can declare the type of `const browserNavIconStates` on line 41 to be `BrowserNavIconStates`

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx (hunk 8 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx]: Removed the comments

File: src/ensembl/src/content/app/browser/Browser.test.tsx (hunk 7 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/Browser.test.tsx]: Updated it here and in few other tests above to use `rerender`

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx (hunk 9 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarControls.test.tsx]: That's done

**Review by azangru — APPROVED** [review]
Looks good
